quic/s2n-quic-transport/src/space/rx_packet_numbers/pending_ack_ranges.rs

```diff
@@ -0,0 +1,284 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{interval_set, space::rx_packet_numbers::ack_ranges::AckRanges};
+use core::time::Duration;
+use s2n_quic_core::{
+    frame::ack::EcnCounts,
+    packet::number::{PacketNumber, PacketNumberRange},
+};
+
+/// Stores ACK ranges pending processing
+#[derive(Clone, Debug, Default)]
+pub struct PendingAckRanges {
+    ranges: AckRanges,
+    ecn_counts: EcnCounts,
+    ack_delay: Duration,
+}
+
+impl PendingAckRanges {
+    #[inline]
+    pub fn new(ranges: AckRanges, ecn_counts: EcnCounts, ack_delay: Duration) -> Self {
+        PendingAckRanges {
+            ranges,
+            ecn_counts,
+            ack_delay,
+        }
+    }
+
+    /// Extend with a packet number range; dropping smaller values if needed
+    #[inline]
+    pub fn extend(
+        &mut self,
+        acked_packets: PacketNumberRange,
```

Comment:
each ACK frame has multiple ranges so this needs to be an `Iterator<Item = PacketNumberRange>`